### ISSUE TITLE
A Less CSS guard written with no space before its condition is still read as ordinary CSS

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#150]
Found while reviewing the fix for #125, which taught `isStandardSyntaxSelector` to recognise a Less CSS guard. It recognises most of them; two shapes still read as ordinary CSS, and the rules go on checking a construct that is none of their business.

### What minimal example or steps are needed to reproduce the bug?

A guard written with no space in front of its condition:

```less
.a:hover when(1 = 1), .b { color: red; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-less",
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/selector-list-comma-newline-after": "always"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

Nothing to report, and the file left alone — as happens the moment a space is put in front of the condition.

### What actually happened?

```
1:21  Expected newline after "," (@stylistic/selector-list-comma-newline-after)
```

and `--fix` writes a line break into the middle of the guard:

```less
.a:hover when(1 = 1),
 .b { color: red; }
```

The same file with `when (1 = 1)` is reported by nothing and left alone. So what the plugin does to a Less stylesheet turns on a space in front of a parenthesis.

Two shapes reach it, both of them a guard written tight:

```less
.a:hover when(1 = 1) { color: red; }
.a:hover when not(1 = 1) { color: red; }
```

There is a second, narrower shape, which comes from the other side. `isStandardSyntaxSelector` reads the guard past any quoted attribute value, since one of those may hold the same text; an escaped quote *outside* a string opens a run that never closes where it should, and the guard behind it is blanked along with it:

```less
.x\'y:hover when ('z' = 'w') { color: red; }
```

```
1:17  Unexpected " ('z' = 'w')" (@stylistic/selector-descendant-combinator-no-non-space)
```

The same guard without the escaped quote in the class name is passed over as it should be.

### Do you have a proposal to fix the bug?

For the first shape, the guard is matched by

```js
/\swhen\s+(?:not\s+)?\(/iu
```

and `\s+` is what asks for the space. `\s*` would take both shapes in, and nothing valid with it: a parenthesis opening in a selector outside a string and outside `:pseudo(` is not CSS at all, and quoted text is blanked before the test. Measured against `.when`, `when`, `.foo when .bar`, `.a.when`, `a:not(.when)`, `.a:nth-child(2n of .when)`, `[data-when="x"]`, `[title=" when ("]` and `.when(x)` — the loose form matches none of them.

What is missing is the one thing that cannot be settled from inside this repository: **whether Less accepts a guard written tight**. If it does, the change above is right. If it does not, `.a:hover when(1 = 1)` is not a guard at all, there is no gap, and what the code wants instead is a test saying so. Either answer takes one run of the Less compiler, or one look at the `guard` production of its parser.

The second shape asks for `withoutQuotedText` to know about CSS escapes outside strings, so that `\'` cannot open one. That is a larger change to a function whose whole point is to be simple, and it is worth doing only if Less accepts such an identifier in the first place — also unverified.

Neither shape is a regression: both read as ordinary CSS before #125 as well. What changed is only that a guard reaching the rules now has somewhere to be noticed.
